test/lib/mayaUsd/render/vp2RenderDelegate/testVP2RenderDelegateUSDPreviewSurface.py

```diff
@@ -247,6 +247,29 @@ def testImportDisplayColor(self):
         # Snapshot and assert similarity
         self.assertSnapshotClose('DisplayColorCube.png')
 
+    def testDoubleSided(self):
+        """
+        Test that backfaces get culled if the doubleSided attribute
+        is enabled on USD prims, and that they don't if it is not.
+        """
+        cmds.file(new=True, force=True)
+        mayaUtils.loadPlugin("mayaUsdPlugin")
+
+        testFile = testUtils.getTestScene("doubleSided", "UsdPreviewSurface.usda")
+        stageShapeNode, stage = mayaUtils.createProxyFromFile(testFile)
+
+        mayaUtils.setBasicCamera(20)
+        self.assertSnapshotClose('doubleSided_enabled_front.png')
+        mayaUtils.setBasicCamera(3)
+        self.assertSnapshotClose('doubleSided_enabled_back.png')
+
+        cubePrim = stage.GetPrimAtPath("/Cube1")
+        cubePrim.GetAttribute('doubleSided').Set(False)
+
+        mayaUtils.setBasicCamera(20)
+        self.assertSnapshotClose('doubleSided_disabled_front.png')
+        mayaUtils.setBasicCamera(3)
+        self.assertSnapshotClose('doubleSided_disabled_back.png')
```

Comment:
The test code for UsdPreviewSurface and MaterialX is 99% the same, however I don't really see where it could be sensibly extracted. It's not really a utility, they just happen to have the same steps. I'd have put both in a dedicated test file for double-sidedness, but MaterialX is optional and its tests separate, so I split it in two.